climada/hazard/base.py

```diff
@@ -1410,3 +1413,142 @@ def _read_att_excel(self, file_name, var_names):
         self.intensity = sparse.csr_matrix(dfr.values[:, 1:num_events + 1].transpose())
         self.fraction = sparse.csr_matrix(np.ones(self.intensity.shape,
                                                   dtype=np.float))
+
+    def extend(self, haz_list, centroids=None, threshold=100):
+        """
+        Extends hazard with events of same type.
+
+        Centroids of all hazards must have the same CRS.
+
+        By default (centroids=None), the centroids of all hazards
+        are combined together. All raster centroids are converted
+        to points and raster data is discarded.
+
+        If centroids is not None, all hazards are defined on these centroids.
+        Centoids of hazards in haz_src not in centroids are mapped onto the
+        nearest point. If a point is further than threshold from the closet
+        centroid, the concatenation fails.
+
+        Note: centroids is modified in place and raster information is
+        destroyed.
+
+
+        Parameters
+        ----------
+        haz_list: list(climada.hazard.Hazard())
+            Hazard instances of the same hazard type
+        centroids: climada.hazard.Centroids(), optional
+            Centroids instance on which to map the concatenated hazard.
+            Default is None.
+        threshold: int or float
+            Threshold for mapping centoids of hazards in haz_list not in
+            centroids.
+            Argument is passed to climada.util.coordinates.assign_coordinates.
+            Default is 100.
+
+        Returns
+        -------
+        None.
+
+        """
+        if self.event_id.any(): #if self is not empty, concatenate with self.
+            haz_list = [self] + haz_list
+        self.__dict__ = self.concatenate_hazard(haz_list,
+                                                centroids,
+                                                threshold).__dict__
```

Comment:
Replacing the `__dict__` object looks mighty dangerous to me. Are we _sure_ nothing unpredicted can happen here? Is there no proper way of re-initialization?

Reply:
Well, I am not sure. Would looping through the dictionary and replacing the values be more sensible?

Reply:
Not really, I'm afraid. I was rather thinking of a constructor that takes arguments and initializes data beyond their mere announcement.
Which we don't have at the moment.

However I'm still uneasy about the `extend` method. And it is partly related to the way the `__dict__` is transferred. To illustrate:
`Hazard.concatenate_hazard([Drought(), Landslide()])` is a `Hazard` object. That's fine.
`Drought().extend([Landslide()])` on the other hand is working on a `Drought` object but any `Drought` property that is not inherited from `Hazard` evaporates. In other words: `extend` bears the hazard of making `Hazard` extensions inconsistent.

Reply:
Ah yes, I understand now. It should actually not be allowed to mix events from different types of hazards. I will add a check for the haz type. Also, the suggestion from below will be implemented.